src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx

```diff
@@ -61,147 +77,215 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
+        }, 100)
+    }
+
+    const preFetchData = (status = '',firstMessageReceived = false) => {
+        const _terminal = terminal
+        _terminal?.reset()
+        // const _fitAddon = fitAddon
+        
+        _terminal.write('Creating pod.')
+        if(status === 'Running'){
+            _terminal.write(' \u001b[38;5;35mSucceeded\u001b[0m')
+            _terminal.writeln('')
+            _terminal.write('Connecting to pod terminal.')
+        }else if(status === 'Failed'){
+            _terminal.write('Failed')
+            _terminal.write(' | \u001b[38;5;110m\u001b[4mCheck Pod Events\u001b[0m')
+            _terminal.write(' | ')
+            _terminal.write('\u001b[38;5;110m\u001b[4mCheck Pod Manifest\u001b[0m')
+        }else{
+            _terminal.write('..')
+        }
+        
+        if(firstMessageReceived){
+            _terminal.write(' \u001b[38;5;35mSucceeded\u001b[0m')
+            _terminal.write(' | \u001b[38;5;110m\u001b[4mCheck Pod Events\u001b[0m')
+            _terminal.write(' | ')
+            _terminal.write('\u001b[38;5;110m\u001b[4mCheck Pod Manifest\u001b[0m')
+            _terminal.writeln('')
+        }else if(status === 'Running'){
+            _terminal.write('..')
+        }
+    }
 
     useEffect(() => {
+        // Maintaining value in ref for setTimeout context
+        socketConnectionRef.current = terminalViewProps.socketConnection
+
         if (terminalViewProps.socketConnection === SocketConnectionType.DISCONNECTING) {
+            if (clustertimeOut) {
+                clearTimeout(clustertimeOut)
+            }
             if (socket) {
-                socket.close();
-                socket = undefined;
+                socket.close()
+                socket = undefined
             }
         }
         if (terminalViewProps.socketConnection === SocketConnectionType.CONNECTING) {
-            getNewSession();
+            getNewSession()
         }
-    }, [terminalViewProps.socketConnection]);
+    }, [terminalViewProps.socketConnection, terminalViewProps.terminalId])
 
     useEffect(() => {
         ReactGA.event({
             category: 'Terminal',
             action: `Selected Pod`,
             label: `${terminalViewProps.nodeName}/${terminalViewProps.containerName}/${terminalViewProps.shell.value}`,
-        });
+        })
 
-        reconnect();
-    }, [terminalViewProps.nodeName]);
+        if (!terminalViewProps.clusterTerminal) {
+            reconnect()
+        } else {
+            terminal?.reset()
+        }
+    }, [terminalViewProps.nodeName])
 
     useEffect(() => {
         ReactGA.event({
             category: 'Terminal',
             action: `Selected Container`,
             label: `${terminalViewProps.nodeName}/${terminalViewProps.containerName}/${terminalViewProps.shell.value}`,
-        });
+        })
 
-        reconnect();
-    }, [terminalViewProps.containerName]);
+        if (!terminalViewProps.clusterTerminal) {
```

Comment:
we can add terminalViewProps.shell dependency in this one so that we don't create new effect for the same